src/codesyncd/handlers/tabs_handler.ts

```diff
@@ -0,0 +1,125 @@
+import path from "path";
+
+import { glob } from "glob";
+
+import { ITabYML } from "../../interface";
+import { generateSettings } from "../../settings";
+import { readYML } from "../../utils/common";
+import { getRandomIndex, getTabsBeingProcessed } from "../utils";
+import { removeFile } from "../../utils/file_utils";
+import { TAB_FILES_PER_ITERATION, TAB_SIZE_LIMIT } from "../../constants";
+import { CodeSyncLogger } from "../../logger";
+import { TabValidator } from "../validators/tab_validator";
+import { TabHandler } from "./tab_handler";
+
+export class TabsHandler {
+    tabYmlFiles : ITabYML[];
+    accessToken: string;
+
+    settings: any;
+    configJSON: any;
+    configRepo: any;
+
+    constructor(repoTab: ITabYML[] | null = null, accessToken: string | null = null) {
```

Comment:
If both params can be null, what's the point for constructor? Maybe should consider defining Static Method that does not need a constructor to be called